src/main/java/com/kine/Kine/food/service/FoodService.java

```diff
@@ -0,0 +1,106 @@
+        Page<Food> foodPage = foodRepository.findAll(pageable);
+        return foodPage.map(Food::ConvertToGETFoodDTO);
+    }
+
+    public Page<GetFoodDTO> findFoodByName(String name, Pageable pageable) {
+        Page<Food> foundFood = foodRepository.findFoodByName(name, pageable);
+        return foundFood.map(Food::ConvertToGETFoodDTO);
+    }
+
+    public GetFoodDTO findFoodById(long id) throws RessourceNotFoundException {
+        Optional<Food> foundFood = foodRepository.findFoodById(id);
+        if(foundFood.isEmpty()){
+            throw new RessourceNotFoundException("Food with ID " + id + " not found");
+        }
+        return foundFood.get().ConvertToGETFoodDTO();
+    }
+
+    public CreateFoodDTO createFood(CreateFoodDTO createFoodDTO) throws RessourceAlreadyExistException, InvalidDataException{
+
+        Food food = createFoodDTO.ConvertToFood();
+        // Check if a food item with the same details already exists
+        if (foodRepository.existsByNameAndCaloriesAndProteinsAndCarbsAndFat(food.getName(), food.getCalories(), food.getProteins(), food.getCarbs(), food.getFat())){
+            throw new RessourceAlreadyExistException("Food already exists");
+        }
+        if (isNotCaloriesRelationValid(food)) {
+            throw new InvalidDataException("Invalid data");
+        }
+        return foodRepository.save(food).ConvertToCreateFoodDTO();
+    }
+
+    public FoodDTO updateFood(Long id, UpdateDTO newFoodData) throws RessourceNotFoundException, RessourceAlreadyExistException, InvalidDataException {
+
+        Optional<Food> existingFoodOptional = foodRepository.findFoodById(id);
+
+        if (existingFoodOptional.isEmpty()){
+            throw new RessourceNotFoundException("Food with ID "  + id +  " not found");
+        }
+
+        Food existingFood = existingFoodOptional.get();
+
+        // Update only the fields that are provided in the newFoodData object
+        if (newFoodData.getName().isPresent()) {
+            existingFood.setName(newFoodData.getName().get());
+        }
+        if (newFoodData.getCalories().isPresent()) {
+            existingFood.setCalories(newFoodData.getCalories().get());
+        }
+        if (newFoodData.getProteins().isPresent()) {
+            existingFood.setProteins(newFoodData.getProteins().get());
+        }
+        if (newFoodData.getCarbs().isPresent()) {
+            existingFood.setCarbs(newFoodData.getCarbs().get());
+        }
+        if (newFoodData.getFat().isPresent()) {
+            existingFood.setFat(newFoodData.getFat().get());
+        }
+
+        if (isNotCaloriesRelationValid(existingFood)) {
+            throw new InvalidDataException("Invalid data");
+        }
+        if (foodRepository.existsByNameAndCaloriesAndProteinsAndCarbsAndFat(existingFood.getName(), existingFood.getCalories(), existingFood.getProteins(), existingFood.getCarbs(), existingFood.getFat())){
+            throw new RessourceAlreadyExistException("Food already exists");
```

Comment:
If it already exists, should we merge them or alert the user ?

Reply:
We can merge them by deleting the one the user is trying to modify. However, if we want to give the user the option, they can choose to remove it themselves.

Reply:
That's a good approach, just say so in the error message. Tell them to either delete this one or use the other

Reply:
That's a good approach, just say so in the error message. Tell them to either delete this one or use the other